File: conformance/bundles/12-combining-marks/greet.steps.rs
//! Rust sibling of `greet.steps.ts` (bundle `12-combining-marks`).

use varar::Steps;

pub fn register(s: &mut Steps<()>) {
    // One slot: echoing the capture back makes the core compare it against the
    // document, which is what exercises the combining-mark span offsets.
    s.sensor("I greet {string}", |_ctx: (), name: String| Ok(name));
}

pub fn state() {}

File: conformance/bundles/12-combining-marks/greet.steps.go
// Go sibling of greet.steps.ts (bundle 12-combining-marks).
package fixture

import "github.com/varar-dev/varar/go/varar"

func Register(s *varar.Steps[varar.Value]) {
	// One slot: echoing the capture back makes the core compare it against the
	// document, which is what exercises the combining-mark span offsets.
	s.Sensor("I greet {string}", func(state varar.Value, name string) (string, error) {
		return name, nil
	})
}

func State() varar.Value {
	return varar.NullValue
}
